tests/bgp/test_4-byte_asn_community.py

```diff
@@ -0,0 +1,231 @@
+                    peer_group_v4 = v['peer group']
+                elif v['ip_version'] == 6:
+                    neigh_ip_v6 = k
+                    peer_group_v6 = v['peer group']
+            assert v['state'] == 'established'
+            neigh_asn[v['description']] = v['remote AS']
+            neighbors[v['description']] = nbrhosts[v['description']]["host"]
+
+    dut_ip_v4 = tbinfo['topo']['properties']['configuration'][neigh]['bgp']['peers'][dut_asn][0]
+    dut_ip_v6 = tbinfo['topo']['properties']['configuration'][neigh]['bgp']['peers'][dut_asn][1]
+
+    dut_ip_bgp_sum = duthost.shell('show ip bgp summary')['stdout']
+    neigh_ip_bgp_sum = nbrhosts[neigh]["host"].shell('show ip bgp summary')['stdout']
+    with open(bgp_id_textfsm) as template:
+        fsm = textfsm.TextFSM(template)
+        dut_bgp_id = fsm.ParseText(dut_ip_bgp_sum)[0][0]
+        neigh_bgp_id = fsm.ParseText(neigh_ip_bgp_sum)[1][0]
+
+    dut_ipv4_network = duthost.shell("show run bgp | grep 'ip prefix-list'")['stdout'].split()[6]
+    dut_ipv6_network = duthost.shell("show run bgp | grep 'ipv6 prefix-list'")['stdout'].split()[6]
+    neigh_ipv4_network = nbrhosts[neigh]["host"].shell("show run bgp | grep 'ip prefix-list'")['stdout'].split()[6]
+    neigh_ipv6_network = nbrhosts[neigh]["host"].shell("show run bgp | grep 'ipv6 prefix-list'")['stdout'].split()[6]
+
+    setup_info = {
+        'duthost': duthost,
+        'neighhost': neighbors[neigh],
+        'neigh': neigh,
+        'dut_asn': dut_asn,
+        'neigh_asn': neigh_asn[neigh],
+        'asn_dict':  neigh_asn,
+        'neighbors': neighbors,
+        'namespace': namespace,
+        'dut_ip_v4': dut_ip_v4,
+        'dut_ip_v6': dut_ip_v6,
+        'neigh_ip_v4': neigh_ip_v4,
+        'neigh_ip_v6': neigh_ip_v6,
+        'peer_group_v4': peer_group_v4,
+        'peer_group_v6': peer_group_v6,
+        'asic_index': asic_index,
+        'dut_bgp_id': dut_bgp_id,
+        'neigh_bgp_id': neigh_bgp_id,
+        'dut_ipv4_network': dut_ipv4_network,
+        'dut_ipv6_network': dut_ipv6_network,
+        'neigh_ipv4_network': neigh_ipv4_network,
+        'neigh_ipv6_network': neigh_ipv6_network
+    }
+
+    logger.info("DUT BGP Config: {}".format(duthost.shell("show run bgp", module_ignore_errors=True)['stdout']))
+    logger.info("Neighbor BGP Config: {}".format(nbrhosts[neigh]["host"].shell("show run bgp")['stdout']))
+    logger.info('Setup_info: {}'.format(setup_info))
+
+    yield setup_info
+
+    # restore config to original state
+    config_reload(duthost)
+    config_reload(neighbors[neigh], is_dut=False)
+
+    # verify sessions are established
+    bgp_facts = duthost.bgp_facts(instance_id=asic_index)['ansible_facts']
+    for k, v in bgp_facts['bgp_neighbors'].items():
+        if v['description'].lower() not in skip_hosts:
+            logger.debug(v['description'])
+            assert v['state'] == 'established'
+
+
+def test_4_byte_asn_community(setup):
+    cmd = 'vtysh -n {} \
```

Comment:
if this is a single-asic linecard, the `namespace` here would be `None`, thus `vtysh -n None` for single-asic cards will have issue

Reply:
use this `run_vtysh` function instead: https://github.com/sonic-net/sonic-mgmt/blob/a6ecc4f3345c481080a296d607b25ee62f5f6d67/tests/common/devices/sonic_asic.py#L399

e.g. `show_res = duthost.asic_instance().run_vtysh("-c 'show version'")`

Reply:
I adjusted the namespace variable to accommodate this.  I did test on single asic and the original form of the command (-n None) did not have a negative impact on the test.

Reply:
can you rename `namespace` to be `cli_options` to avoid naming confusion?
example of usage: 
`cli_options = " -n {}".format(namespace) if namespace is not None else ""`

Reply:
I changed the variable to avoid confusion.